apps/src/templates/progress/ProgressLesson.jsx

```diff
@@ -105,12 +107,13 @@ class ProgressLesson extends React.Component {
 
     const showAsLocked = isLockedForUser || isLockedForAllStudents;
 
-    const title = lesson.lessonNumber
-      ? i18n.lessonNumbered({
-          lessonNumber: lesson.lessonNumber,
-          lessonName: lesson.name,
-        })
-      : lesson.name;
+    const title =
+      lesson.lessonNumber && !unitHasUnnumberedLessons
+        ? i18n.lessonNumbered({
+            lessonNumber: lesson.lessonNumber,
+            lessonName: lesson.name,
+          })
+        : lesson.name;
```

Comment:
I'm not switching to lesson.localized_title here because the logic is subtly different. I'm not sure which logic is right, but I settled on "do no harm" and preserved existing behavior for numbered lessons in this PR.

Reply:
here is the logic which computes the lesson numbers that get used here: https://github.com/code-dot-org/code-dot-org/blob/98b6379cbaae9e434403f91bd142db3d76a6e6f9/apps/src/code-studio/progressRedux.ts#L569-L589